src/terminal/terminal.rs

```diff
@@ -658,6 +670,73 @@ where
         Ok(())
     }
 
+    /// Implement `Self::insert_before` using scrolling regions.
+    #[cfg(feature = "scrolling-regions")]
+    fn insert_before_scrolling_regions(
```

Comment:
Doc comment could be a little bit more detailed. Ideally enough to explain why this is here and how it is different from the previous method. Or put another way, enough to make it easy not to have to read the method implementation to work out what's happening. Nothing huge, just 1-3 extra sentences.

Reply:
Okay, I updated it.